src/main/kotlin/org/rust/lang/core/types/infer/Expressions.kt

```diff
@@ -136,6 +144,19 @@ fun inferExpressionType(expr: RsExpr): Ty {
     }
 }
 
+private fun inferTypeForExprAndDefaultMethodImpl(expr: RsMethodCallExpr, method: RsFunction, type: RsTypeReference?): Ty {
+    val traitItem = method.parentOfType<RsTraitItem>() ?: return TyUnknown
+    val caller = expr.expr as? RsPathExpr ?: return TyUnknown
+    val typeStruct = caller.type as? TyStruct ?: return TyUnknown
+    val project = PsiUtilCore.getProjectInReadAction(typeStruct.item)
+    val impls = RsImplIndex.findImpls(project, typeStruct)
```

Comment:
It's better not to call into `RsImplIndex` directly, but to use `findImplsAndTraits` function, because it handles stuff like bounds on type parameters.